beta-src/src/state/game/game-api-slice.ts

```diff
@@ -308,6 +308,52 @@ const drawBuilds = (state) => {
   }
 };
 
+const updateUnitsDisbanding = (state) => {
+  const {
+    data: {
+      data: { contextVars, currentOrders },
+    },
+    territoriesMeta,
+    ordersMeta,
+    overview: { phase },
+  }: {
+    data;
+    territoriesMeta: TerritoriesMeta;
+    ordersMeta;
+    overview: {
+      phase: GameOverviewResponse["phase"];
+      members: GameOverviewResponse["members"];
+    };
+  } = current(state);
+  if (currentOrders && contextVars && territoriesMeta) {
+    if (phase === "Retreats") {
+      const userDisbandingUnits = currentOrders.filter(
+        (o) =>
+          ordersMeta[o.id].update.type === "Disband" || o.type === "Disband",
+      );
```

Comment:
You're looping over `currentOrders` here, remember `ordersMeta` gets updated throughout the user's session, not `currentOrders`. Current orders only gets updated on initial map load or when we make additional API calls to game/data.